awx/main/management/commands/inventory_import.py

```diff
@@ -75,7 +75,14 @@ def get_base_args(self):
         bargs.extend(['-v', '{0}:{0}:Z'.format(self.source)])
         for key, value in STANDARD_INVENTORY_UPDATE_ENV.items():
             bargs.extend(['-e', '{0}={1}'.format(key, value)])
-        bargs.extend([get_default_execution_environment().image])
+        ee = get_default_execution_environment()
+        bargs.extend([ee.image])
+        authpath = ee.build_authfile(functioning_dir(self.source))
```

Comment:
really going to need some cleanup.

I have found `tempfile.NamedTemporaryFile` to be preferable. Normally, this would not be an option because uWSGI processes are somewhat long-lived, but in this case `awx-manage` invocations will exit when the work is done, so cleanup would happen at the right time.

Reply:
This is functionality shared with code in tasks.py.  Would a temp file be appropriate there as well?  Should the temp file still wind up in the relevant directory in each case, or /tmp/ or somewhere else like that?  And what are the implications of trying to do that in a container filesystem?

Reply:
`self.source` is provided by the user by the `--source` option, right? That is quite different from `tasks.py`.

The user may not want you to mess with the directory where their inventory is.

I don't see any need to put the authfile anywhere near the inventory file. You probably want to _avoid_ putting it in the container mounted directories, if possible.